src/main/java/org/opensearch/security/securityconf/SecurityRoles.java

```diff
@@ -97,5 +97,13 @@ Set<String> getAllPermittedIndicesForDashboards(
 
     SecurityRoles filter(Set<String> roles);
 
+    SecurityRoles createSecurityRole(
```

Comment:
I am introducing this to future-proof this change so that it can be expanded on in the future to assign additional perms to a plugin user.

Practically, this will be used to create an "in-memory" role that can be used for the pluginUser so that existing Privilege Evaluation can be entirely re-used. 

One thing I did have to do was duplicate the SystemIndexRegistry from core to maintain a map of plugin identifiers (the canonical class name) -> set of registered system index patterns.

The in-memory role takes advantage of the `system:admin/system_index` permission introduced in https://github.com/opensearch-project/security/pull/2887 where the "in-memory" role will effectively be:

```
<plugin_identifier>:
  cluster_permissions:
    - "indices:data/write/bulk" // bulk is a cluster perm. A bulk will fail if one of the items in a bulk request fails.
  index_permissions:
    - index_patterns:
        - ".my-system-index"
        - ...
      allowed_actions:
        - "*"
        - "system:admin/system_index"
 ```